### ISSUE TITLE
Javadocs output is entirely broken for method summary

### ISSUE THREAD

**Issue by LambdAurora** [asiekierka/MinecraftPOMF#2429]
Method summary on classes is entirely broken as the lines are not complete and then the content get shifted.

Here you can see it on a live javadoc: https://maven.fabricmc.net/docs/yarn-21w20a+build.9/net/minecraft/entity/passive/AnimalEntity.html

and a screenshot:
![image](https://user-images.githubusercontent.com/12587332/119350054-27e3e200-bc9f-11eb-9f86-a05e6447d84d.png)

This happens for any classes.

**Comment by liach** [comment]
See openjdk/jdk#4066. Needa wait for java 17 to get this fix

**Comment by modmuss50** [comment]
Could try building against 17? idk if gradle 7 supports that?

**Comment by liach** [comment]
> 
> 
> Could try building against 17? idk if gradle 7 supports that?

A few things to consider:
1. The current jdk 17 ea build 22 doesn't include that fix yet; need to wait for build 23. Then we can use github `setup-java` action to get ea build. See https://github.com/actions/setup-java#supported-version-syntax
2. If build 23 includes [strongly encapsulating jdk internals by default](https://openjdk.java.net/jeps/403), I expect it to break gradle, or even a significant portion of minecraft mods; I actually wonder if the reflection hack for de-finaling fields can still work with that. It is possible that openjdk pr 4015, which implements this somehow scary feature, may be included in build 23

**Comment by modmuss50** [comment]
1. Thats great! no major rush for the fix, but I would like it before jdk17 is here.

2. I believe this is not an issue for us as JDK 16 set it to `--illegal-access=deny` that JEP is going to `change the treatment of the permit, warn, and debug modes so that they have no effect,`

This all all annoyingly blocked by gradle not even atempting to run on jdk 17: https://github.com/modmuss50/yarn/runs/2656968544?check_suite_focus=true

https://github.com/gradle/gradle/issues/16857

Idk what it would take to get gradle running on 17, might not be a huge effort and we definally have a number of uses for it. Another option is to backport the fix to JDK 16 and build it into a docker image.

**Comment by liach** [comment]
For that log, it appears that gradle is using an older groovy that does not use asm 9.1; asm 9.1 (which most of our toolchain is already using) supports java 17's class files. We need to figure out a way for gradle to ship newer groovy as a result.

**Comment by modmuss50** [comment]
Seems like the easyist solution will be to use a toolchain when building the javadoc. Gradle doesnt seem to have a nice way to download custom JDK builds for everyone, but can use locally installed JDK's passed in via ENV vars such as 'JDK17' or via the gradle propertry `org.gradle.java.installations.paths`.

JDK 17 Build 23 was released today, but does not seem to have your fix, so this is on hold utill build 24, its not a major issue.

**Comment by liach** [comment]
@modmuss50 We can consider downloading development builds from https://builds.shipilev.net/openjdk-jdk/ hosted by a jdk contributor aleksey shipilyov. However, I am not good at bash for properly decompressing these binaries or adding them to gradle variables, so it's up to you for using this or waiting for ea. Maybe we can try using older jdk to do javadoc as the generated source and package info are theoretically compatible with older java.

**Comment by modmuss50** [comment]
I would perfer to use a source I trust, im sure those builds are fine but its not something we need that badly. (If it was id build it myself). Lets wait for build 24

**Comment by liach** [comment]
Hmm, build 24 is out, but I am not sure how to make gradle's java toolchain and github action's java installation work

**Comment by modmuss50** [comment]
Great, I can handle that. Will take a bit of changing to it.
